projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php

```diff
@@ -220,6 +221,44 @@ public function callback( $path = '', $blog_id = 0 ) {
 			$query['has_password'] = false;
 		}
 
+		$formatted_top_posts = array();
+
+		if ( isset( $args['order_by'] ) && 'views' === $args['order_by'] ) {
+			$date      = gmdate( 'Y-m-d' );
+			$days      = 30;
+			$max_posts = isset( $args['number'] ) ? $args['number'] : 20;
+
+			$stats_history_query = stats_get_daily_history( false, $blog_id, 'postviews', 'post_id', $date, $days, '', $max_posts + 1, true );
```

Comment:
Since `stats_get_daily_history` is a WordPress.com-only function, we'll need to add a safeguard to avoid fatal errors on self-hosted sites.

Reply:
Nice, [safe guarded the code here](https://github.com/Automattic/jetpack/pull/29095/commits/20916129ba58ee732c82bf1db756fa7e1a9f8884).

Reply:
Is it okay that WoA sites will not have access to this functionality in this case?

Reply:
Good point @jeherve, using this strategy we'll only support simple sites, not ATOMIC or self-hosted sites. Started a thread here to understand if we'll move this strategy forward: p1677198597447039-slack-CKZHG0QCR

Thanks for your review!

Reply:
Thanks again for your reviews @jeherve, we changed our gears and decided to move forward [using this strategy](https://github.com/Automattic/jetpack/pull/29149). Could you review it?